satpy/tests/test_composites.py

```diff
@@ -485,8 +489,10 @@ def test_night_only_area_with_alpha(self):
             comp = DayNightCompositor(name="dn_test", day_night="night_only", include_alpha=True)
             res = comp((self.data_b,))
             res = res.compute()
-        expected_l_channel = np.array([[np.nan, 0.], [0.5, 1.]])
-        expected_alpha = np.array([[np.nan, 0.], [0., 0.]])
+        expected_l_channel = np.array([[np.nan, 0.], [0.49999994, 1.]], dtype=np.float32)
+        # FIXME: with the current changes the np.nan becomes 0.0 instead, why?!
+        expected_alpha = np.array([[np.nan, 0.], [0., 0.]], dtype=np.float32)
```

Comment:
The `np.nan` will be changed to `0.0` with the PR. No idea why, so didn't update the test.

Reply:
Ok, just needed a simple update to getting the mask after changing the "missing" data from integer 0 to `data.dtype`.